### ISSUE TITLE
v5: export utilities such as EventHandler and SelectorEngine

### ISSUE THREAD

**Issue by mistic100** [twbs/bootstrap#32941]
I maintain https://github.com/mistic100/Bootstrap-Confirmation which is an abstraction above Popover to add contextual buttons.

While removing jQuery dependency I find myself having to copy at lot of utility code from Boostrap such as `EventHandler`, `SelectorEngine` etc.

It would be much useful for plugins developpers to be able to import such utilities from Boostrap itself instead of duplicating code.

**Comment by mistic100** [comment]
This is particularly import for the "Data" helper.

Because the data are not actually stored on the HTMLElement but on a private singleton store, there is not way for a plugin to access it.

**Comment by XhmikosR** [comment]
I don't think we ever planned on these to be part of the public API. Perhaps the Data helper might make sense, but the rest not so much AFAICT.

We can export more if needed in a minor release as long as there's consensus it's useful.

**Comment by mistic100** [comment]
I think it would be a very good idea to promote a coherent plugin architecture for v5. Previous versions benefited from jQuery for this, it would be a shame the ecosystem splits into a multitudes of API and usages.

**Comment by XhmikosR** [comment]
Feel free to make a PR. Just keep in mind we really need to get the regressions fixed first so that we can release v5 stable, so any help there is more than welcome.

**Comment by mistic100** [comment]
If I do a PR it will be only to export the mentionned utils. You might aggree that this subject needs a bit more of planning and conception from the core team.

**Comment by XhmikosR** [comment]
Yeah, for sure. It can wait until someone else chimes in too.

**Comment by bewards** [comment]
What would the reason be for not making SelectorEngine accessible? It seems like it has some useful helper functions for finding elements with a pattern that works in IE11.

**Comment by BigOHenry** [comment]
Hello, any progress on this?
